src/api.js

```diff
@@ -924,6 +926,11 @@ class VLWholeSlideMicroscopyImageViewer {
       customDrawOptions.style = options.style;
     }
     const allDrawOptions = Object.assign(defaultDrawOptions, customDrawOptions);
+
+    if (options.condition) {
+      allDrawOptions.condition = options.condition;
```

Comment:
What type has `condition`? Is it [ol/events/condition/Contition](https://openlayers.org/en/latest/apidoc/module-ol_events_condition.html#~Condition)? We don't want to expose the underlying Openlayers API. Can we abstract this in a meaningful way?

Reply:
Hey, its actually just a vanilla JS callback that takes the open layers event and returns a boolean, its part of the `Draw` api in `ol`. Here is an example from the app I'm currently building, to set a tool on left click and pan on middle click:

```js
activateTool(toolName) {
  this.viewer.deactivateModifyInteraction();
  this.viewer.activateDrawInteraction({
    geometryType: toolName,
    condition: function(e) {
      // Only draw on left click
      if (e.pointerEvent.buttons === 1) {
        return true;
      } else {
        return false;
      }
    },
  });

  this.viewer.activateDragPanInteraction({
    condition: function(e) {
      // Only pan on middle click
      if (e.pointerEvent.buttons === 4) {
        return true;
      } else {
        return false;
      }
    },
  });
}
```

However, the event is synthetic, so I guess it requires minimal `ol` knowledge.
I'm not sure there is any other use for the callback other than filtering for bindings + modifier keys, so we could have the surface api take something like:

```
bindings: {
  mouseButtons: ['left','right'],
  modifierKey: 'ctrl'
}
```

And then generate a corresponding condition in an `ol` compatible within this function call, and pass this to `ol` `Draw`? The default would be no condition if `bindings` wasn't present?

There is also touch support one may have to worry about, and unfortunately this isn't something my client requires at this time. An api like above could be expanded when someone wants this though?

Cheers

Reply:
I like the idea of the surface API for `bindings`. This is a nice level of abstraction that we should be able to translation to another library if that would every become necessary.